sewer/dns_providers/tests/test_gandi.py

```diff
@@ -0,0 +1,154 @@
+from unittest import mock
+from unittest import TestCase
+
+import sewer
+
+MOCK_GANDI_API_KEY = 'gandi-Api-Key'
+
+class MockResponseObject:
+    def __init__(self, status_code=200, body=None):
+        self.status_code = status_code
+        self.body = body
+
+    def json(self):
+        return self.body
+
+
+def add_side_effect_to_request_get(mock_requests_get):
+    def mock_implementation(*args, **kwargs):
+        assert kwargs['headers'] == {"X-Api-Key": MOCK_GANDI_API_KEY}, 'Get headers not set'
+
+        if args[0] == 'https://dns.api.gandi.net/api/v5/domains/second-level-domain.tld':
+            return MockResponseObject(body={
+                'zone_records_href': 'mock_zone_records_href'
+            })
+
+        if args[0] == 'mock_zone_records_href':
+            return MockResponseObject(body=[
+                {
+                    'rrset_href': 'mock_record_href',
+                    'rrset_type': 'TXT',
+                    'rrset_ttl': 10800,
+                    'rrset_name': '_acme-challenge.subsubdomain.subdomain',
+                    'rrset_values': [
+                        'challenge_text'
+                    ]
+                }
+            ])
+
+        return MockResponseObject(status_code=404)
+
+    mock_requests_get.side_effect = mock_implementation
+
+
+def add_side_effect_to_request_post(mock_requests_post):
+    def mock_implementation(*args, **kwargs):
+        assert kwargs['headers'] == {
+            "X-Api-Key": MOCK_GANDI_API_KEY,
+            "Content-Type": "application/json"},\
+            'Post headers not set'
+
+        if args[0] == 'mock_zone_records_href':
+            return MockResponseObject()
+
+        return MockResponseObject(status_code=404)
+
+    mock_requests_post.side_effect = mock_implementation
+
+
+def add_side_effect_to_request_delete(mock_requests_delete):
+    def mock_implementation(*args, **kwargs):
+        assert kwargs['headers'] == {"X-Api-Key": MOCK_GANDI_API_KEY}, 'Get headers not set'
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/komuW/sewer/pullRequest?prid=5350192)